[6877] Add an OpenAPI specification to the API documentation (#4260)
DFE-Digital/register-trainee-teachers: app/views/api_docs/reference/v0.1/_reference.md

This API allows you to access information about trainees and provides endpoints to update and create trainee data.

<div class="govuk-inset-text">
  Open API specifications can be viewed and downloaded here:
  <a href="/api-docs/v0.1/openapi" target="_blank">Open API</a>
</div>

> 

Reviewer: I think this might sit better in the 'Developing on the API' section where the reference spreadsheet is.

Author: Like this?
<img width="657" alt="Screenshot 2024-05-13 at 12 14 33" src="https://github.com/DFE-Digital/register-trainee-teachers/assets/6339025/1cc2486e-9c37-446c-8973-1bd80c7b4b16">

Download [Register API reference spreadsheet v0.1 (Excel)](/api-docs/reference/Register_API_Reference_v0.1.xlsx)

### OpenAPI

> An OpenAPI schema can be viewed and downloaded here: <a href="/api-docs/v0.1/openapi" target="_blank">Open API</a>

Reviewer: Let's go with similar wording to Apply:
```suggestion
The OpenAPI spec for this API is <a href="/api-docs/v0.1/openapi" target="_blank">available in YAML format</a>.
```
